### PR TITLE
SSCS-10581 admin action correction remake PR

### PR DESCRIPTION
**Pull request by amoslv1**
**Before creating a pull request make sure that:**

- [ ] commit messages are meaningful and follow good commit message guidelines
- [ ] README and other documentation has been updated / added (if needed)
- [ ] tests have been updated / new tests has been added (if needed)

Please remove this line and everything above and fill the following sections:

### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ ] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/adminactioncorrection/AdminActionCorrectionAboutToStartHandler.java [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+
+        PreSubmitCallbackResponse<SscsCaseData> response = new PreSubmitCallbackResponse<>(sscsCaseData);
+
+        String caseId = sscsCaseData.getCcdCaseId();
+        if (!SscsUtil.isSAndLCase(sscsCaseData)) {
+            log.error("Admin Action Correction: Cannot process non Scheduling & Listing Case for Case ID {}", caseId);
+            response.addError("Cannot process Admin Action Correction on non Scheduling & Listing Case");
+            return response;
+        }
+
```
Comment (jarekPierchala) [inline src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/adminactioncorrection/AdminActionCorrectionAboutToStartHandler.java:49]: no need for that empty line

File: src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/adminactioncorrection/AdminActionCorrectionAboutToSubmitHandler.java [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+        log.info("Handling admin correction for case: {}", ccdCaseId);
+        AdminCorrectionType adminCorrectionType = sscsCaseData.getPostHearing().getCorrection().getAdminCorrectionType();
+
+        if (AdminCorrectionType.HEADER.equals(adminCorrectionType)) {
+            log.info("Handling header correction for case: {}", ccdCaseId);
+            SscsUtil.handleFinalDecisionNotice(sscsCaseData, preSubmitCallbackResponse, previewDocumentService, decisionNoticeService);
+
+        } else if (AdminCorrectionType.BODY.equals(adminCorrectionType)) {
+            log.info("Sending body correction to judge for case: {}", ccdCaseId);
+
```
Comment (jarekPierchala) [inline src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/adminactioncorrection/AdminActionCorrectionAboutToSubmitHandler.java:58]: no need for that empty line

File: src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/adminactioncorrection/AdminActionCorrectionMidEventHandler.java [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+            YesNo noticeGenerated = sscsCaseData.getFinalDecisionNoticeGenerated();
+
+            if (isYes(noticeGenerated)) {
+                // IF generated: automatically regenerate final decision with the current details
+                log.info("Admin Action Correction: Regenerating final decision for caseId {}", caseId);
+                String templateId = documentConfiguration.getDocuments()
+                    .get(sscsCaseData.getLanguagePreference()).get(DECISION_ISSUED);
+                preSubmitCallbackResponse = issueDocument(callback, DECISION_NOTICE, templateId, generateFile, userAuthorisation);
+            } else {
+                // IF uploaded: go to upload screen and expect user to upload new document
```
Comment (jarekPierchala) [inline src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/adminactioncorrection/AdminActionCorrectionMidEventHandler.java:72]: should that be done? if not and something will be added here as part of other ticket add TODO

File: src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/adminactioncorrection/AdminActionCorrectionSubmittedCallbackHandler.java [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+
+        PreSubmitCallbackResponse<SscsCaseData> response = new PreSubmitCallbackResponse<>(caseData);
+
+        Long caseId = Long.valueOf(caseData.getCcdCaseId());
+        PostHearing postHearing = caseData.getPostHearing();
+        AdminCorrectionType adminCorrectionType = postHearing.getCorrection().getAdminCorrectionType();
+        log.info("Admin Action Correction: handling adminActionCorrection {} for case {}", adminCorrectionType,  caseId);
+
+        CcdCallbackMap callbackMap = postHearing.getCorrection().getAdminCorrectionType();
+
```
Comment (jarekPierchala) [inline src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/adminactioncorrection/AdminActionCorrectionSubmittedCallbackHandler.java:53]: no need for that empty line

**Review by jarekPierchala — CHANGES_REQUESTED** [review]

File: src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/adminactioncorrection/AdminActionCorrectionAboutToStartHandler.java [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+            && callback.getEvent() == EventType.ADMIN_ACTION_CORRECTION
+            && isPostHearingsEnabled;
+    }
+
+    @Override
+    public PreSubmitCallbackResponse<SscsCaseData> handle(CallbackType callbackType, Callback<SscsCaseData> callback, String userAuthorisation) {
+
+        final CaseDetails<SscsCaseData> caseDetails = callback.getCaseDetails();
+        final SscsCaseData sscsCaseData = caseDetails.getCaseData();
+
```
[the remark this answers is not in the document]
Comment (amoslv1) [reply src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/adminactioncorrection/AdminActionCorrectionAboutToStartHandler.java:40]: done

File: src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/adminactioncorrection/AdminActionCorrectionAboutToStartHandler.java (hunk 1 above)
Comment (amoslv1) [reply src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/adminactioncorrection/AdminActionCorrectionAboutToStartHandler.java:49]: done

File: src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/adminactioncorrection/AdminActionCorrectionAboutToSubmitHandler.java [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+        String ccdCaseId = sscsCaseData.getCcdCaseId();
+
+        log.info("Handling admin correction for case: {}", ccdCaseId);
+        AdminCorrectionType adminCorrectionType = sscsCaseData.getPostHearing().getCorrection().getAdminCorrectionType();
+
+        if (AdminCorrectionType.HEADER.equals(adminCorrectionType)) {
+            log.info("Handling header correction for case: {}", ccdCaseId);
+            SscsUtil.handleFinalDecisionNotice(sscsCaseData, preSubmitCallbackResponse, previewDocumentService, decisionNoticeService);
+
```
[the remark this answers is not in the document]
Comment (amoslv1) [reply src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/adminactioncorrection/AdminActionCorrectionAboutToSubmitHandler.java:55]: done

File: src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/adminactioncorrection/AdminActionCorrectionAboutToSubmitHandler.java (hunk 2 above)
Comment (amoslv1) [reply src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/adminactioncorrection/AdminActionCorrectionAboutToSubmitHandler.java:58]: done

File: src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/adminactioncorrection/AdminActionCorrectionMidEventHandler.java [hunk 7]
```diff
@@ -0,0 +1,79 @@
...
+        log.info("Admin Action Correction: handling callback with pageId {} for caseId {}", pageId, caseId);
+
+        PreSubmitCallbackResponse<SscsCaseData> preSubmitCallbackResponse =
+            new PreSubmitCallbackResponse<>(sscsCaseData);
+        AdminCorrectionType adminCorrectionType = sscsCaseData.getPostHearing().getCorrection().getAdminCorrectionType();
+
+        if (isNull(adminCorrectionType)) {
+            log.error(String.format("adminCorrectionType unexpectedly null for case: %s", caseId));
+            preSubmitCallbackResponse.addError(String.format("adminCorrectionType unexpectedly null for case: %s", caseId));
+
```
[the remark this answers is not in the document]
Comment (amoslv1) [reply src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/adminactioncorrection/AdminActionCorrectionMidEventHandler.java:60]: done

File: src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/adminactioncorrection/AdminActionCorrectionMidEventHandler.java (hunk 3 above)
Comment (amoslv1) [reply src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/adminactioncorrection/AdminActionCorrectionMidEventHandler.java:72]: I'm not sure if there is any code required here to instigate this functionality, or if it is entirely caused by ccd defs. Will add TODO to that effect anyway.

File: src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/adminactioncorrection/AdminActionCorrectionSubmittedCallbackHandler.java [hunk 8]
```diff
@@ -0,0 +1,64 @@
...
+            && callback.getEvent().equals(EventType.ADMIN_ACTION_CORRECTION)
+            && isPostHearingsEnabled;
+    }
+
+    @Override
+    public PreSubmitCallbackResponse<SscsCaseData> handle(CallbackType callbackType, Callback<SscsCaseData> callback,
+        String userAuthorisation) {
+
+        SscsCaseData caseData = callback.getCaseDetails().getCaseData();
+
```
[the remark this answers is not in the document]
Comment (amoslv1) [reply src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/adminactioncorrection/AdminActionCorrectionSubmittedCallbackHandler.java:44]: done

File: src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/adminactioncorrection/AdminActionCorrectionSubmittedCallbackHandler.java (hunk 4 above)
Comment (amoslv1) [reply src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/adminactioncorrection/AdminActionCorrectionSubmittedCallbackHandler.java:53]: done

File: src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/adminactioncorrection/AdminActionCorrectionSubmittedCallbackHandler.java [hunk 9]
```diff
@@ -0,0 +1,64 @@
...
+        CcdCallbackMap callbackMap = postHearing.getCorrection().getAdminCorrectionType();
+
+        if (isNull(callbackMap)) {
+            response.addError("Invalid Admin Correction Type Selected or correction "
+                + "selected as callback is null");
+            return response;
+        }
+
+        caseData = ccdCallbackMapService.handleCcdCallbackMap(callbackMap, caseData);
+
```
[the remark this answers is not in the document]
Comment (amoslv1) [reply src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/adminactioncorrection/AdminActionCorrectionSubmittedCallbackHandler.java:61]: done

**Review by hajam09 — DISMISSED** [review]

**Pull request closed without merging**